process_pr.py

```diff
@@ -1517,7 +1517,7 @@ def process_pr(repo_config, gh, repo, issue, dryRun, cmsbuild_user=None, force=F
         missing_commits = set(
             k for k in bot_cache["commits"] if not bot_cache["commits"][k].get("squashed", False)
         ).difference(all_commit_shas)
-        last_seen_commit_time = None
+        new_commits = all_commit_shas.difference(k for k in bot_cache["commits"])
```

Comment:
why not move it under the condition when there are no differences between last seen and new commits i.e under https://github.com/cms-sw/cms-bot/pull/2144/files#diff-f37d46a488510bd0b96fd201ff189f8a704b756502df6a1cd46b117986e608a4R1550 ?